Mission7/schema.sql

```diff
@@ -0,0 +1,67 @@
+-- 사용자(users Table)
+CREATE TABLE users (
+  id SERIAL PRIMARY KEY,
+  email VARCHAR(255) NOT NULL UNIQUE,
+  nickname VARCHAR(100) NOT NULL,
+  password TEXT NOT NULL,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+);
+
+-- 게시글(article Table)
+CREATE TABLE article (
+  id SERIAL PRIMARY KEY,
+  title VARCHAR(255) NOT NULL,
+  content TEXT NOT NULL,
+  user_id int,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+  FOREING KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
+);
+
+-- 제품(product)
+CREATE TABLE product (
+  id  SERIAL PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  description TEXT NOT NULL,
+  price INT NOT NULL,
+  tags TEXT[] NOT NULL,
+  user_id int,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+  FOREING KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
+)
+
+-- 댓글(commend)
+CREATE TABLE commend (
+  id SERIAL PRIMARY KEY,
+  content TEXT NOT NULL,
+  product_id int,
+  article_id int,
+  user_id int,
+  created_at TIMESTAMP DEFAULT NOW(),
+  updated_at TIMESTAMP DEFAULT NOW(),
+  FOREING KEY (user_id) REFERENCES users(id) ON DELETE CASCADE,
+  FOREING KEY (product_id) REFERENCES product(id) ON DELETE CASCADE,
+  FOREING KEY (article_id) REFERENCES article(id) ON DELETE CASCADE
+)
+
+-- 즐겨찾기(favorite)
+CREATE TABLE favorite (
+  id SERIAL PRIMARY KEY,
+  product_id int UNIQUE,
+  user_id int UNIQUE,
+  FOREING KEY (product_id) REFERENCES product(id) ON DELETE CASCADE,
+  FOREING KEY (user_id) REFERENCES users(id) ON DELETE CASCADE
+)
```

Comment:
아래 like와 마찬가지로 다대다 관계 표현하기 위해서 다음과 같이 만들어야합니다. 프로덕트와 유저는 각각 유니크하면 안되고 테이블내 여러개 있을 수 있고 (product, user) 이런 pair가 유일해야하므로 복합 Primary key를 만들면 됩니다.
```suggestion
CREATE TABLE user_favorite_products (
  user_id INTEGER NOT NULL REFERENCES users(id) ON DELETE CASCADE,
  product_id INTEGER NOT NULL REFERENCES products(id) ON DELETE CASCADE,
  PRIMARY KEY (user_id, product_id) -- 복합 PK로 중복 방지
);
```